src/main.rs

```diff
@@ -1,16 +1,104 @@
-use anyhow::{ Result};
-use cli_lang_tool::{Dictionary, Word};
+use anyhow::{Result};
+use clli::{Dictionary, Word};
 use std::{fs::File, io::{prelude::*, BufReader}, path::PathBuf};
 use serde::{Serialize, Deserialize};
 use rand::Rng;
-use structopt::StructOpt;
+use clap::{ Parser, Subcommand};
 
 
 const FILE_PATH: &str = "./examples/storage/turk.json";
 
+#[derive(Debug, Parser)]
+#[clap(name = "clli")]
+#[clap(about = "turkish lang tool", long_about = None)]
+struct Cli {
+    #[clap(subcommand)]
+    command: Commands,
+}
+
+#[derive(Debug, Subcommand)]
+enum Commands {
+    // #[clap(arg_required_else_help = true)]
+    ///Get the random word from the sozluk
+    Random{
+        /// count - how many random words
+        #[clap(default_value_t = 1)]
+        count: u8,
+    },
+    #[clap(arg_required_else_help = true)]
+    /// Adds records to the dictionary
+    Add
+        {
+        #[clap(required = true)]
+        native: String,
+        english: String,
+        },
+
+    /// Get the count of words in the dictionary 
+    Count,
+
+    /// Get the all words from the dictionaryy
+    GetAllWords,
+
+    /// Translate english -> turkish
+    Etranslate{
+        #[clap(required = true)]
+        english: String,
+    },
+
+    /// Translate turkish -> english 
+    Ttranslate{
+        #[clap(required = true)]
+        native: String,
+    },    
+}
 
 fn main() {
-    println!("Hello, sozluk");
+    let args = Cli::parse();
+    let mut sozluk = sozluk_init();
+
+    match args.command {
+        Commands::Random{ count}  => {
+            for _ in 0..count {
+                if let Some(word) = sozluk.get_rand_word() {
+                    println!("The random word : {}. translate:{}", &word.native, &word.english );
+                    }
+                }
+            }
+        Commands::Add { native , english } => {
+            println!("Pushing to {} :: {}", native , english);
+            let dummy_cat: Vec<String> = vec![];
+            let dummy_examples: Vec<String> = vec![];
```

Comment:
Why do we have these guys here? They can be passed into the function directly.

Reply:
Stub. To replace in the future with real vec![String, String, String, String, ].